lib/src/compiler.dart

```diff
@@ -307,6 +307,8 @@ abstract class AbstractCompiler {
       filepath = 'dart/$filepath';
     } else if (uri.scheme == 'file') {
       filepath = path.relative(filepath, from: inputBaseDir);
+      // Play nice with paths not under inputBaseDir (which may be ''):
```

Comment:
can you explain the thinking here?
do we need to offer inputBaseDir as an option?